webapp/app/forms/fields.py

```diff
@@ -15,18 +15,48 @@
 from app.forms.validators import ValidElsterCharacterSet
 
 
+class NumericInputMixin:
+
+    @staticmethod
+    def set_inputmode(**kwargs):
+        kwargs.setdefault('inputmode', 'numeric')
+        kwargs.setdefault('pattern', '[0-9]*')
+
+        return kwargs
+
+
 class SteuerlotseStringField(StringField):
 
     def pre_validate(self, form):
         ValidElsterCharacterSet().__call__(form, self)
 
 
+class SteuerlotseIntegerField(NumericInputMixin, IntegerField):
+
+    def __call__(self, *args, **kwargs):
+        kwargs = self.set_inputmode(**kwargs)
+
+        return super().__call__(**kwargs)
+
+
+class SteuerlotseNumericStringField(NumericInputMixin, StringField):
+
+    def __call__(self, *args, **kwargs):
+        kwargs = self.set_inputmode(**kwargs)
+
+        return super().__call__(**kwargs)
+
+
 class MultipleInputFieldWidget(TextInput):
     """A divided input field."""
     sub_field_separator = ''
     input_field_lengths = []
     input_field_labels = []
 
+    @staticmethod
+    def set_inputmode(**kwargs):
+        return kwargs
+
```

Comment:
It confused me that we need the method here. I get why you added it. However, I think I'd make more sense to call `set_inputmode` only in the classes that also derive from `NumericInputMixin`, hence Date and IdNr. So these two would have a method like this:

```
def __call__(self, *args, **kwargs):
        kwargs = self.set_inputmode(**kwargs)

        return super().__call__(**kwargs)
```

Reply:
Good catch. You are completely right, we do not need this. We could also put the call method you are describing into NumericInputMixin.

That said, we should maybe also talk about this, because I would like to change the Mixin a little bit because of the EuroField.

Reply:
Ok, we've implemented the described idea